### PR TITLE
LESQ-1030: Parent subjects utilised if exist

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1965

- Utilises parent lesson for curriculum links if the parent subject exists
- Creates function to convert subject title to string
- Amends test / fixture data

## Issue(s)

Fixes #LESQ-1030

## Acceptance Criteria:

- [x]  Child subjects should link to the equivalent parent subject page (for the example above you should be taken to secondary science aqa)

## How to test

1. Go to https://deploy-preview-2719--oak-web-application.netlify.thenational.academy
2. Click on all units should navigate to the correct pages, ks4 subjects with parent science go to science pages

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/unitListing/convertSubjectToSlug.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export function convertSubjectToSlug(subject: string): string {
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/unitListing/convertSubjectToSlug.ts:1]: Had this refactored to be a map of Subjects and corresponding slugs, however, function could return undefined which was throwing so strange type errors further down the line.

**Review by rightsaidjames — APPROVED** [review]
Works as it should for KS4 Science, all other unit listing pages unaffected, thanks!

File: src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts [hunk 2]
```diff
@@ -19,9 +19,13 @@ const unitListingQuery =
       return null;
     }
 
-    const parsedRawUnits = unitsForProgramme.map((p) =>
-      syntheticUnitvariantLessonsSchema.parse(p),
-    );
+    // For some reason the parse is parsing out subject parent
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts:22]: what does this mean?

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts:22]: I can't access the student_parent after the `syntheticUnitvariantLessonsSchema.parse(p)` it gets filtered out the programme_fields obj

Comment (k-huggs) [reply src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts]: fixed ✅

File: src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts [hunk 3]
```diff
@@ -32,10 +33,13 @@ const unitListingQuery =
 
     const programmeFields = firstUnit.programme_fields;
 
+    console.log(programmeFields.subject_parent, "< Subject Parent");
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/unitListing/unitListing.query.ts:36]: 👀

**Review by njblore — APPROVED** [review]

**Pull request closed without merging**